Fullscreen control element

Alyxion opened the pull request:
I recently created a component to make use Quasars's AppFullscreen plugin and thought it were also helpful for other users, so I documented it and added it to the NiceGUI docu.

It enables the user to switch to fullscreen programmatically which especially on a smartphone or tablet context greatly improves the user experience.

![image](https://github.com/user-attachments/assets/b55be2a6-785c-4a46-8740-30811a3626da)

falkoschindler reviewed (approved):
Thanks a lot for this pull request, @Alyxion!

I finally found the time to review your code. Apart from several minor changes here and there, I decided to derive `Fullscreen` from `ValueElement`. This enables us to re-use plenty of code and improves consistency with other elements, adds binding support etc.

I also renamed `ui.app_fullscreen` into `ui.fullscreen` for simplicity.

From my point of view the PR is ready to merge. Do you agree? Does the modified implementation work for you?

Alyxion commented:
> Thanks a lot for this pull request, @Alyxion!
> 
> I finally found the time to review your code. Apart from several minor changes here and there, I decided to derive `Fullscreen` from `ValueElement`. This enables us to re-use plenty of code and improves consistency with other elements, adds binding support etc.
> 
> I also renamed `ui.app_fullscreen` into `ui.fullscreen` for simplicity.
> 
> From my point of view the PR is ready to merge. Do you agree? Does the modified implementation work for you?

Thank you very much for the changes, they are fine of course. Originally it was actually named ui.fullscreen but I renamed it as I didn't know how close to the Quasar names you want to keep the NiceGUI imlementations. But yes, also like ui.fullscreen more anyways.
